src/main/java/org/opensearch/sdk/rest/SDKHttpRequest.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.sdk.rest;
+
+import org.opensearch.common.bytes.BytesReference;
+import org.opensearch.extensions.rest.ExtensionRestRequest;
+import org.opensearch.http.HttpRequest;
+import org.opensearch.http.HttpResponse;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.RestStatus;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class helps to get instance of HttpRequest
+ */
+public class SDKHttpRequest implements HttpRequest {
+    private final RestRequest.Method method;
+    private final String uri;
+    private final BytesReference content;
+    private final Map<String, List<String>> headers;
+    private final HttpVersion httpVersion;
+
+    /**
+    * Instantiates this class with a copy of {@link ExtensionRestRequest}
+    *
+    * @param request The request
+    */
+    public SDKHttpRequest(ExtensionRestRequest request) {
+        this.method = request.method();
+        this.uri = request.uri();
+        this.content = request.content();
+        this.headers = request.headers();
+        this.httpVersion = request.httpVersion();
+    }
+
+    @Override
+    public RestRequest.Method method() {
+        return method;
+    }
+
+    @Override
+    public String uri() {
+        return uri;
+    }
+
+    @Override
+    public BytesReference content() {
+        return content;
+    }
+
+    @Override
+    public Map<String, List<String>> getHeaders() {
+        return headers;
+    }
+
```

Comment:
We should add tests for `SDKHttpRequest` and test these getters too.

Reply:
> We should add tests for `SDKHttpRequest` and test these getters too for better code coverage.

Good point, @owaiskazi19 .

The HttpRequest is embedded internally into the RestRequest, so the existing uses of `createTestRestRequest()` inside the `TestBaseExtensionRestHandler` instantiate both of these.   We should probably:
 - create a new test class `TestSDKRestRequest`
 - Move that static `createTestRestRequest()` method there so other test methods that need it can call it
 - add some tests of all the (implemented) getters.  Most tests of the SDKHttpRequest will come from getters on SDKRestRequest, for example the `RestRequest` `method()` and `uri()` methods just delegate to the internal `HttpRequest`.